esp-hal/src/timer/systimer.rs

```diff
@@ -463,11 +430,17 @@ where
             .bit_is_set();
 
         let us = value.ticks();
-        let ticks = us * (SystemTimer::TICKS_PER_SECOND / 1_000_000) as u64;
+        let ticks = us * (SystemTimer::TICKS_PER_SECOND / 1_000_000);
 
         if auto_reload {
             // Period mode
 
+            // The `SYSTIMER_TARGETx_PERIOD` field is 26-bits wide, so we must
```

Comment:
on S2 it's bits 0:29

I wonder if we could make use of the `width()` in the PACs `FieldWriter`?